lib/dao.js

```diff
@@ -1683,7 +1683,27 @@ DataAccessObject._coerce = function(where, options) {
     }
     var DataType = props[p] && props[p].type;
     if (!DataType) {
-      continue;
+      // Handle nested properties
```

Comment:
Please consider replacing this comment with calling a named function extracted from the `if` body.

```js
if (!DataType && p.indexOf('.') !== -1) {
  DataType = this._coerceNestedProperty(p/*, ...*/);
} 

if (!DataType) {
  continue;
}
```